addons/l10n_in_withholding/wizard/l10n_in_account_withhold_wizard.py

```diff
@@ -0,0 +1,303 @@
+    _check_company_auto = True
+
+    @api.model
+    def default_get(self, fields_list):
+        result = super().default_get(fields_list)
+        active_model = self._context.get('active_model')
+        active_ids = self._context.get('active_ids')
+        if len(active_ids) > 1:
+            raise UserError(_("You can only create a withhold for only one record at a time."))
+        if active_model not in ('account.move', 'account.payment') or not active_ids:
+            raise UserError(_("TDS must be created from an Invoice or a Payment."))
+        active_record = self.env[active_model].browse(active_ids)
+        result['reference'] = _("TDS of %s", active_record.name)
+        if active_model == 'account.move':
+            if active_record.move_type not in ('out_invoice', 'out_refund', 'in_invoice', 'in_refund') or active_record.state != 'posted':
+                raise UserError(_("TDS must be created from Posted Customer Invoices, Customer Credit Notes, Vendor Bills or Vendor Refunds."))
+            result['related_move_id'] = active_record.id
+            result['type_name'] = active_record.type_name
+        elif active_model == 'account.payment':
+            display_map = active_record._get_aml_default_display_map()
+            type_name = display_map.get((active_record.payment_type, active_record.partner_type))
+            if not active_record.partner_id:
+                raise UserError(_("Please set a partner on the %s before creating a withhold.", type_name))
+            result['related_payment_id'] = active_record.id
+            result['type_name'] = type_name
+        return result
+
+    reference = fields.Char(string="Reference")
+    type_name = fields.Char(string="Type")
+    withhold_type = fields.Selection([
+        ('in_withhold', "Purchase"),
+        ('out_withhold', "Sale"),
+        ('in_refund_withhold', "Purchase Refund"),
+        ('out_refund_withhold', "Sale Refund"),
+    ], string="Withholding Type", required=True, compute='_compute_withhold_type', store=True, precompute=True)
+    related_move_id = fields.Many2one(
+        comodel_name='account.move',
+        string="Invoice/Bill",
+        readonly=True,
+    )
+    related_payment_id = fields.Many2one(
+        comodel_name='account.payment',
+        string="Payment",
+        readonly=True,
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string="Partner",
+        compute='_compute_partner_id'
+    )
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string="Company",
+        compute='_compute_company_id'
+    )
+    display_amount_warning = fields.Boolean(
+        string="Display Amount Warning",
+        compute='_compute_display_amount_warning',
+    )
+    journal_id = fields.Many2one(
+        comodel_name='account.journal',
+        string="Journal",
+        compute='_compute_journal',
+        precompute=True,
+        readonly=False,
+        store=True,
```

Comment:
```suggestion
        compute='_compute_journal', precompute=True,
        readonly=False, store=True,
```
or something that brings similar stuff a bit together